mitiq/pec/representations/learning.py

```diff
@@ -172,28 +302,67 @@ def biased_noise_loss_function(
     """
     epsilon = params[0]
     eta = params[1]
-    representations = [
-        represent_operation_with_local_biased_noise(
-            operation,
-            epsilon,
-            eta,
-        )
-        for operation in operations_to_mitigate
-    ]
-    mitigated_values = np.array(
-        [
-            execute_with_pec(
-                circuit=training_circuit,
-                observable=observable,
-                executor=noisy_executor,
-                representations=representations,
-                full_output=False,
-                **pec_kwargs,
+
+    if pec_data.size > 0:
+        ind_eps = np.abs(pec_data[:, 0, 0] - epsilon).argmin()
+        ind_eta = np.abs(pec_data[0, :, 0] - eta).argmin()
+        mitigated_values = pec_data[ind_eps, ind_eta, :]
+
+    else:
+        representations = [
+            represent_operation_with_local_biased_noise(
+                operation,
+                epsilon,
+                eta,
             )
-            for training_circuit in training_circuits
+            for operation in operations_to_mitigate
         ]
-    )
+        mitigated_values = np.array(
+            [
+                execute_with_pec(
+                    circuit=training_circuit,
+                    observable=observable,
+                    executor=noisy_executor,
+                    representations=representations,
+                    full_output=False,
+                    **pec_kwargs,
+                )
+                for training_circuit in training_circuits
+            ]
+        )
 
     return np.mean(
         abs(mitigated_values.reshape(-1, 1) - ideal_values.reshape(-1, 1)) ** 2
     )
+
+
+def _parse_learning_kwargs(
+    learning_kwargs: Dict["str", Any]
+) -> Tuple[npt.NDArray[np.float64], str, Dict["str", Any]]:
+    r"""Function for handling additional options and data for the learning
+    functions.
+
+    Args:
+        learning_kwargs: Additional data and options including
+            ``pec_data`` from pre-executed runs of PEC on training circuits,
+            ``method`` an optimization method supported by
+            ``scipy.optimize.minimize``, and settings for the chosen
+            optimization method.
+
+    Returns:
+       Values contained in learning_kwargs or defaults if not specified.
+    """
+    minimize_kwargs = learning_kwargs.get("learning_kwargs")
+
+    if minimize_kwargs is not None:
+        pec_data = minimize_kwargs.pop(
+            "pec_data", np.array([])
+        )  # type: ignore
+        method = minimize_kwargs.pop("method", "Nelder-Mead")  # type: ignore
```

Comment:
Is there a reason for using `pop` here? Seems strange to mutate the `learning_kwargs` object.

Reply:
It is a bit strange, but the idea is to obscure `pec_data` from the user. We have to get it out somehow because it's not an "option" related to the chosen `method` in `minimize`. 
The `method` is another matter- it could remain in `minimize_kwargs` except that we want Nelder-Mead to be default (not the default for `minimize`) because that's what seems to work best for this problem so far.